source/CargoHold.cpp

```diff
@@ -595,3 +595,23 @@ int CargoHold::IllegalCargoFine() const
 
 	return totalFine;
 }
+
+
+
+// Returns the amount tons of illegal cargo.
+int CargoHold::IllegalCargoAmount() const
+{
+	int count = 0;
+
+	// Find any illegal outfits inside the cargo hold.
+	for(const auto &it : outfits)
+		if(it.first->Get("illegal") || it.first->Get("atrocity") > 0.)
+			count += it.second * it.first->Mass();
```

Comment:
> Looks like the new formula won't catch zero-mass illegal outfits like Nerve Gas.

For the purposes of IllegalCargoAmount, should we use a minimum mass of 1? i.e. make the following change:
```suggestion
			count += it.second * max(1., it.first->Mass());
```
To my understanding, the logic behind this change is that, assuming zero scan interference, if your cargo hold is 1/3rd illegal goods or greater then you'll always be caught because anyone glancing at your cargo could spot that much illegal goods, but if you have way more legal good than illegal goods then it becomes harder to spot the illegal goods since they're hidden among the legal goods. So even though nerve gas or any potential other H2H outfit doesn't have a mass for the purposes of your ship, it still takes up some volume in your cargo hold, hence the suggestion to have a minimum mass of 1 for the purposes of this function.

I think that masking your illegal goods with legal goods sounds like a way more interesting mechanic than just throwing on more interference plating, although it admittedly seems questionable in terms of the in-universe explanation since when the player does a cargo scan it returns a list of all cargo on the scanned ship, and if it's the exact same for an NPC scanning you then regardless of how many legal goods you have, they'll be able to spot the illegal one. For the purposes of gameplay though I'm fine with handwaving it.

Reply:
I like the minimum, but I would make a distinction between installed outfits and outfits in cargo.

Installed outfits must be ready to go and be used at a moment's notice by the crew, and so would be stored in more accessible and thus detectable areas such as a ship's weapons lockers.

Cargo in outfits can't be used because it's, for example, buried under a mountain of coal. This would make it less detectable at the cost of being unusable.

As such, I'd keep a minimum of counts-as-one-ton for *installed* H2H weapons, but if they're kept in cargo it only counts as 0.25 tons.

Reply:
Note that this PR only applies to cargo, so we don't need to worry about distinguishing between installed and in-cargo outfits in terms of their "size" on scans.